app/services/supplementary-billing/abstraction-billing-period.service.js

```diff
@@ -46,6 +46,8 @@ function go (billingPeriod, chargePurpose) {
 
   _flagPeriodsForConsideration(billingPeriod, abstractionPeriods)
 
+  _calculateBillablePeriods(abstractionPeriods, billingPeriod)
+
   _calculateBillableDays(abstractionPeriods)
```

Comment:
My suggestion would be that we don't bother to call these methods if the abstraction period is not to be considered. IMHO it's a waste of CPU cycles. That would mean some refactoring.

One way would be rather than `_calculateBillablePeriods()` and `_calculateBillableDays()` taking the whole array, instead you would iterate through `abstractionPeriods` and where a period is flagged for consideration, just pass it through to these methods (which would then be refactored to work on a single instance).

Or, use [filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) to extract a subset of periods to be considered and just pass that through to these methods.

Reply:
Agreed. I guess given that this function could potentially be getting called many many times per bill run it would make sense to make it as efficient as possible.

Reply:
Yep, that would defo do the trick as well. Probably clearer that we're only returning out of the service stuff that needs dealing with, so the calling service doesn't have to do any filtering of its own.